openshift/caddy-reverse-proxy.dc.yaml

```diff
@@ -0,0 +1,119 @@
+---
+apiVersion: v1
+kind: Template
+labels:
+  template: "${APP_NAME}-reverse-proxy-template"
+  app: "${APP_NAME}-reverse-proxy"
+metadata:
+  name: "${APP_NAME}-reverse-proxy-dc"
+objects:
+  # - kind: NetworkSecurityPolicy
+  #   apiVersion: secops.pathfinder.gov.bc.ca/v1alpha1
+  #   metadata:
+  #     name: "${APP_NAME}-reverse-proxy-pods-to-egress-${NAMESPACE}"
+  #   spec:
+  #     description: |
+  #       Allow app pods to open connections to the internet
+  #     source:
+  #       - - "$namespace=${NAMESPACE}"
+  #         - "app=${APP_NAME}-reverse-proxy"
+  #         - "deploymentconfig=${APP_NAME}-reverse-proxy"
+  #         - role=reverse-proxy
+  #     destination:
+  #       - - ext:network=any
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: "${APP_NAME}-caddyfile"
+    data:
+      Caddyfile: |
+        *:2015 {
+            redir 302 {
+                https://comfort.pathfinder.gov.bc.ca/app/${APP_NAME}/
```

Comment:
can't we pass that in as a param?

Reply:
We can template out that string further if it makes sense. However, as this yaml file never had any intent of being a part of the pipeline, there wasn't much need to add another layer of template parameterization. Also by only parameterizing the specific pathname of the form, it makes it easier to get this out and running all at once.

Reply:
It makes the template far more useable without modification.  you can default it for this particular use case if you want, but you already pass a parameter in, so allow it to accept another.